test/e2e/framework/pv/pv.go

```diff
@@ -800,3 +804,52 @@ func SkipIfNoDefaultStorageClass(c clientset.Interface) {
 		e2eskipper.Skipf("error finding default storageClass : %v", err)
 	}
 }
+
+// GetWindowsStorageClassName returns Windows storageClass or return error
+func GetWindowsStorageClassName(c clientset.Interface) (string, error) {
+	list, err := c.StorageV1().StorageClasses().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("Error listing storage classes: %v", err)
+	}
+	var scName string
+	for _, sc := range list.Items {
+		if IsWindowsAnnotation(sc.ObjectMeta) {
+			if len(scName) != 0 {
+				return "", fmt.Errorf("Multiple Windows storage classes found: %q and %q", scName, sc.Name)
+			}
+			scName = sc.Name
+		}
+	}
+	if len(scName) == 0 {
+		return "", fmt.Errorf("No Windows storage class found")
+	}
+	return scName, nil
+}
+
+// SkipIfNoWindowsStorageClass skips tests if no Windows SC can be found.
+func SkipIfNoWindowsStorageClass(c clientset.Interface) {
+	_, err := GetWindowsStorageClassName(c)
+	if err != nil {
+		e2eskipper.Skipf("error finding windows storageClass : %v", err)
+	}
+}
+
+// SkipIfNoStorageClass skips tests if no default SC for linux or Windows SC
+func SkipIfNoStorageClass(c clientset.Interface) {
+	if framework.NodeOSDistroIs("windows") {
+		SkipIfNoWindowsStorageClass(c)
+	} else {
+		SkipIfNoDefaultStorageClass(c)
+	}
+
+}
+
+// IsWindowsAnnotation returns a boolean if
+// the annotation IsWindowsStorageClassAnnotation is set
+func IsWindowsAnnotation(obj metav1.ObjectMeta) bool {
```

Comment:
are we planning to have more of these?

Reply:
For in-tree plugin, we might have to use this solution.
For CSI long-term, we plan to use same default storage class as Linux, but it requires a few changes including API change which will take some time. 
1. change fsType setting behavior: CSI driver will select default fsType based on OS during mounting time instead of always defaulting to ext4 by external-provisioner during volume provisioning time. 
2. PV has a field of fsType in spec: currently pv_controller creates PV object and sets this field when provisioning the volume. Because of the change 1 above, pv_controller cannot set it during provisioning time. However, kubelet and driver do not have the permission to update PV spec which is immutable. We are proposing to add fsType into PV status and update it later during mounting time.